erts/emulator/drivers/common/efile_drv.c

```diff
@@ -1261,26 +1261,61 @@ static void free_read_line(void *data)
     EF_FREE(d);
 }
 
+void read_file_zero_size(struct t_data* d);
+
+/* [ERL-327] Some special files like /proc/... have reported size 0 */
+void read_file_zero_size(struct t_data* d) {
+    size_t actual_size = 0;
+    while (1) {
+        ssize_t read_result;
+        char* datap = 0;
+
+        /* Realloc output binary to be a bit (256k) larger than the data
+         * already in it. This will be the chunk size (FILE_SEGMENT_READ) */
+        d->c.read_file.binp = driver_realloc_binary(
+          d->c.read_file.binp, actual_size + FILE_SEGMENT_READ);
+
+        datap = d->c.read_file.binp->orig_bytes + actual_size;
+
+        /* Read until we hit EOF (read less than requested) */
+        read_result = read((int) d->fd, datap, FILE_SEGMENT_READ);
```

Comment:
You should use the `efile_read(errInfo, flags, fd, buf, count, pBytesRead)` wrapper instead of plain `read(fd, p, size)` since the former for example handles both retry on EINTR as well as uses the right call on Windows.